elasticsearch/src/main/java/io/micronaut/elasticsearch/DefaultElasticsearchClientFactory.java

```diff
@@ -54,20 +64,54 @@ RestClient restClient(DefaultElasticsearchConfigurationProperties elasticsearchC
         return restClientBuilder(elasticsearchConfiguration).build();
     }
 
+    /**
+     * @param transport The {@link ElasticsearchTransport} object.
+     * @return The ElasticsearchClient.
+     * @since 4.2.0
+     */
+    @Singleton
+    ElasticsearchClient elasticsearchClient(ElasticsearchTransport transport) {
+        return new ElasticsearchClient(transport);
+    }
+
+    /**
+     * @param transport The {@link ElasticsearchTransport} object.
+     * @return The ElasticsearchAsyncClient.
+     * @since 4.2.0
+     */
+    @Singleton
+    ElasticsearchAsyncClient elasticsearchAsyncClient(ElasticsearchTransport transport) {
+        return new ElasticsearchAsyncClient(transport);
+    }
+
+    /**
+     * @param elasticsearchConfiguration The {@link DefaultElasticsearchConfigurationProperties} object.
+     * @param objectMapper The {@link ObjectMapper} object.
+     * @return The {@link ElasticsearchTransport}.
+     * @since 4.2.0
+     */
+    @Bean(preDestroy = "close")
```

Comment:
This should be a `@Singleton` too or it won't be shared between either client.

Reply:
done

Reply:
The change breaks the shutdown of the transport. You need both `@Bean` and the scope:

```
 @Bean(preDestroy = "close")
 @Singleton
```

Reply:
done